propulsion-reactor/Program.fs

```diff
@@ -351,26 +368,38 @@ module Args =
             | _, _, Some p, _ ->            Percentage p
             | None, None, None, true ->     StartPos.TailOrCheckpoint
             | None, None, None, _ ->        StartPos.StartOrCheckpoint
-
-        member __.Discovery =
-            match __.Tcp, __.Port with
-            | false, None ->   Discovery.GossipDns            __.Host
-            | false, Some p -> Discovery.GossipDnsCustomPort (__.Host, p)
-            | true, None ->    Discovery.Uri                 (UriBuilder("tcp", __.Host).Uri)
-            | true, Some p ->  Discovery.Uri                 (UriBuilder("tcp", __.Host, p).Uri)
         member __.Tcp =
             a.Contains EsSourceParameters.Tcp
             || EnvVar.tryGet "EQUINOX_ES_TCP" |> Option.exists (fun s -> String.Equals(s, bool.TrueString, StringComparison.OrdinalIgnoreCase))
         member __.Port =                    match a.TryGetResult Port with Some x -> Some x | None -> EnvVar.tryGet "EQUINOX_ES_PORT" |> Option.map int
         member __.Host =                    a.TryGetResult Host     |> defaultWithEnvVar "EQUINOX_ES_HOST"     "Host"
         member __.User =                    a.TryGetResult Username |> defaultWithEnvVar "EQUINOX_ES_USERNAME" "Username"
         member __.Password =                a.TryGetResult Password |> defaultWithEnvVar "EQUINOX_ES_PASSWORD" "Password"
+        member __.ProjTcp =
+            a.Contains EsSourceParameters.ProjTcp
+            || EnvVar.tryGet "EQUINOX_ES_PROJ_TCP" |> Option.exists (fun s -> String.Equals(s, bool.TrueString, StringComparison.OrdinalIgnoreCase))
+        member __.ProjPort =                match a.TryGetResult Port with Some x -> Some x | None -> EnvVar.tryGet "EQUINOX_ES_PROJ_PORT" |> Option.map int
+        member __.ProjHost =                a.TryGetResult Host     |> defaultWithEnvVar "EQUINOX_ES_PROJ_HOST"     "ProjHost"
+        member __.ProjUser =                a.TryGetResult Username |> defaultWithEnvVar "EQUINOX_ES_PROJ_USERNAME" "Username"
+        member __.ProjPassword =            a.TryGetResult Password |> defaultWithEnvVar "EQUINOX_ES_PROJ_PASSWORD" "Password"
         member __.Retries =                 a.GetResult(EsSourceParameters.Retries, 3)
         member __.Timeout =                 a.GetResult(EsSourceParameters.Timeout, 20.) |> TimeSpan.FromSeconds
         member __.Heartbeat =               a.GetResult(HeartbeatTimeout, 1.5) |> TimeSpan.FromSeconds
+        
+        member x.ConnectProj(log: ILogger, storeLog: ILogger, appName, nodePreference) =
+            let s (x : TimeSpan) = x.TotalSeconds
+            let discovery = discovery(x.ProjHost, x.ProjPort, x.ProjTcp)
+            log.ForContext("host", x.ProjHost).ForContext("port", x.ProjPort)
```

Comment:
I'd replace `"host"` with `"projHost"` etc so a log in splunk etc. can be filtered without having to refer to the message (i.e. looking for that starting with `"Projection "` to tell it apart fro the other log entry)

Reply:
makes sense